plugins/scenario/hu.bme.mit.gamma.scenario.statechart.generator/src/hu/bme/mit/gamma/scenario/statechart/generator/StatechartGenerator.xtend

```diff
@@ -480,12 +484,13 @@ class StatechartGenerator extends ScenarioModelSwitch<EObject> {
 			if (set.size == 1 && set.get(0) instanceof NegatedModalInteraction &&
 				(set.get(0) as NegatedModalInteraction).modalinteraction instanceof Signal &&
 				!((set.get(0) as NegatedModalInteraction).modalinteraction as Signal).arguments.empty) {
-				signals = set.filter[set.size == 1].filter[it instanceof NegatedModalInteraction].filter [
+				signals = set.filter[set.size == 1].filter(NegatedModalInteraction).filter [
 					(it as NegatedModalInteraction).modalinteraction instanceof Signal
```

Comment:
These castings are unnecessary as `filter(NegatedModalInteraction)` already takes care of the castings.